app/routes.test.js

```diff
@@ -37,6 +42,55 @@ const checkAuthorisedRouteNotLoggedIn = path => (
       expect(res.headers.location).toEqual('http://identity-server/login');
     }));
 
+const checkForbiddenNoCsrf = path => request(setUpFakeApp())
+  .post(path)
+  .set('Cookie', [mockAuthorisedCookie])
+  .type('form')
+  .send({})
+  .then((res) => {
+    expect(res.status).toEqual(403);
+  });
+
+const checkRedirectToLogin = async (csrfPagePath, postPath) => {
+  const { cookies, csrfToken } = await getCsrfTokenFromGet(
+    setUpFakeApp(), csrfPagePath, mockAuthorisedCookie,
+  );
+  return request(setUpFakeApp())
+    .post(postPath)
+    .type('form')
+    .set('Cookie', [cookies])
+    .send({
+      _csrf: csrfToken,
+    })
+    .expect(302)
+    .then((res) => {
+      expect(res.redirect).toEqual(true);
+      expect(res.headers.location).toEqual('http://identity-server/login');
+    });
+};
+
+const checkLoggedInNotAuthorised = async (csrfPagePath, postPath) => {
+  const { cookies, csrfToken } = await getCsrfTokenFromGet(
+    setUpFakeApp(), csrfPagePath, mockAuthorisedCookie,
+  );
+
+  const mockUnauthorisedJwtPayload = JSON.stringify({
+    id: '88421113', name: 'Cool Dude',
+  });
+  const mockUnauthorisedCookie = `fakeToken=${mockUnauthorisedJwtPayload}`;
+
+  return request(setUpFakeApp())
+    .post(postPath)
+    .type('form')
+    .set('Cookie', [cookies, mockUnauthorisedCookie])
+    .send({ _csrf: csrfToken })
+    .expect(200)
+    .then((res) => {
+      expect(res.text.includes('data-test-id="error-title"')).toEqual(true);
+      expect(res.text.includes('You are not authorised to view this page')).toEqual(true);
+    });
+};
```

Comment:
we really need to move the above 3 into BCL. I'll add to tech-debt.

Reply:
at this point is it worth it?

Reply:
I've added the ticket. If we get round to it then it'll be nice. It just clutters up the routes.test.js with about 30 lines before you get to the main bit